File: agent/src/controllers/docker.controller.ts
import { Request, Response } from "express";
import { DockerService } from "../services/docker.service";

export class DockerController {
  static async getContainers(_: Request, response: Response) {
    try {
      const containers = await DockerService.getContainers();

      response.json(containers);
    } catch (error) {
      response.status(500).json({
        error: "Failed to fetch docker containers",
      });
    }
  }
}

File: agent-go/controllers/docker_controller.go
package controllers

import (
	"net/http"

	"agent-go/services"

	"github.com/gin-gonic/gin"
)

func GetContainers(c *gin.Context) {
	containers, err := services.GetContainers()

	if err != nil {
		c.JSON(http.StatusInternalServerError, gin.H{
			"error": "Failed to fetch containers",
		})

		return
	}

	c.JSON(http.StatusOK, containers)
}